### PR TITLE
Feature/mpi pio.hype1

### PR DESCRIPTION
**Pull request by ShervanGharari**

### CODE REVIEW

**Comment by ShervanGharari** [comment]
this is just a test pull request, Ill remove it later today, no action is needed on this pull request.

**Comment by ShervanGharari** [comment]
The resolved conflict from the pull #216 (which is closed). All the changed from that pull request is present here.

**Comment by ShervanGharari** [comment]
Not able to compile on Graham, I should closer look into this

**Comment by ShervanGharari** [comment]
included the latest development of cesm-coupling branch, able to compile the code on Graham using the tailored makefiles. next is to check if the code result in identical simulation.

**Comment by nmizukami** [comment]
The following external library files are tracked 

route/build/lib/piolib/include/gptl.h
route/build/lib/piolib/include/pio.h
route/build/lib/piolib/include/uthash.h
route/build/lib/piolib/install_manifest.txt
route/build/lib/piolib/lib/libpio.settings
route/build/lib/piolib/libpio.settings

Would you be able to exclude these? 

I will update .gitignore file to prevent git from trying to track those files.

**Comment by ShervanGharari** [comment]
The branch `ShervanGharari:feature/mpi-pio.HYPE1`  compiles successfully on Graham, ComputeCanada.

File: route/build/src/lake_route.f90 [hunk 1]
```diff
@@ -150,6 +172,14 @@ subroutine lake_route(&
...
           ! the model is Doll03
           ! print*, "lake model is Doll 2003"
+
+          ! temporary solution, this should be removed if there is restart activated...
+          print*, iTime, 'iTime in the area'
+          if (iTIme == 1) then
+            RCHFLX_out(iens,segIndex)%REACH_VOL(1) = RPARAM_in(segIndex)%D03_MaxStorage
+          endif
+
+          !
```
Comment (nmizukami) [inline route/build/src/lake_route.f90:182]: I think you can remove this

File: route/build/src/process_ntopo.f90 [hunk 2]
```diff
@@ -419,6 +419,19 @@ subroutine put_data_struct(nSeg, structSEG, structNTOPO, &
...
+     RPARAM_in(iSeg)%HYP_E_emr       =     structSEG(iSeg)%var(ixSEG%HYP_E_emr)%dat(1)
+     RPARAM_in(iSeg)%HYP_E_lim       =     structSEG(iSeg)%var(ixSEG%HYP_E_lim)%dat(1)
+     RPARAM_in(iSeg)%HYP_E_min       =     structSEG(iSeg)%var(ixSEG%HYP_E_min)%dat(1)
+     RPARAM_in(iSeg)%HYP_E_zero      =     structSEG(iSeg)%var(ixSEG%HYP_E_zero)%dat(1)
+     RPARAM_in(iSeg)%HYP_Qrate_emr   =     structSEG(iSeg)%var(ixSEG%HYP_Qrate_emr)%dat(1)
+     RPARAM_in(iSeg)%HYP_Erate_emr   =     structSEG(iSeg)%var(ixSEG%HYP_Erate_emr)%dat(1)
+     RPARAM_in(iSeg)%HYP_Qrate_prim  =     structSEG(iSeg)%var(ixSEG%HYP_Qrate_prim)%dat(1)
+     RPARAM_in(iSeg)%HYP_Qrate_amp   =     structSEG(iSeg)%var(ixSEG%HYP_Qrate_amp)%dat(1)
+     RPARAM_in(iSeg)%HYP_Qrate_phs   =     structSEG(iSeg)%var(ixSEG%HYP_Qrate_phs)%dat(1)
+     RPARAM_in(iSeg)%HYP_prim_F      =     structSEG(iSeg)%var(ixSEG%HYP_prim_F)%dat(1)
```
Comment (nmizukami) [inline route/build/src/process_ntopo.f90:432]: RPARAM_in(iSeg)%HYP_prim_F is logical array.  gnu compiler complains. structSEG(iSeg)%var(ixSEG%HYP_prim_F)%dat(1) -> (structSEG(iSeg)%var(ixSEG%HYP_prim_F)%dat(1)==1).

File: route/build/src/lake_route.f90 (hunk 1 above)
Comment (ShervanGharari) [reply route/build/src/lake_route.f90:182]: @nmizukami this is added together with the `is_vol_wm_jumpstart` flag which easier scenario building with the user. in this case the user needs to changed the target volume and if this flag is on the volume for the lakes and reservoir following a target volume will be exactly the same value as in the target volumes time series. this is however different from initializing the parametric lakes that can be done via restart file. any thoughts suggestion on that?

File: route/build/src/process_ntopo.f90 (hunk 2 above)
Comment (ShervanGharari) [reply route/build/src/process_ntopo.f90]: should have been resolved. please check

File: route/build/src/lake_route.f90 (hunk 1 above)
Comment (ShervanGharari) [reply route/build/src/lake_route.f90]: sorry I confused this with the flag to jump start the state of lake, this is commented and will be removed.

**Comment by ShervanGharari** [comment]
`ESCOMP:cesm-coupling` and `ShervanGharari:feature/mpi-pio.HYPE1` result in identical simulation when lake is turned off.

**Pull request merged**